# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/base/Pivot.h
# Commit: c6db418af190
# Full Hash: c6db418af19097f562d94b87bbf25362556b2fbc
# Author: Morgan Rae Reschenberg <mreschenberg@mozilla.com>
# Date: 2022-10-21 08:39:18
# Regressor Bug: 1787284
# File Overlap Count: 1
# Description:
#   Bug 1787284: [Part 2] Cache name and create radio name pivot rule r=eeejay
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D159118
# ==============================================================================

diff -r fda6706bb580 -r c6db418af190 accessible/base/Pivot.h
--- a/accessible/base/Pivot.h	Thu Oct 20 21:40:05 2022 +0000
+++ b/accessible/base/Pivot.h	Thu Oct 20 21:40:06 2022 +0000
@@ -122,6 +122,20 @@
   virtual uint16_t Match(Accessible* aAcc) override;
 };
 
+/**
+ * This rule matches remote radio button accessibles with the given name
+ * attribute. It assumes the cache is enabled.
+ */
+class PivotRadioNameRule : public PivotRule {
+ public:
+  explicit PivotRadioNameRule(const nsString& aName);
+
+  virtual uint16_t Match(Accessible* aAcc) override;
+
+ protected:
+  const nsString& mName;
+};
+
 }  // namespace a11y
 }  // namespace mozilla
 